### DISCUSSION TITLE
How can we determine which port was chosen by the kernel when starting mitmproxy with port=0?

### DISCUSSION THREAD

**Discussion by acuozzo** [mitmproxy/mitmproxy#5978] [Q&A] [answered]
I'm running mitmproxy via DumpMaster in Python. How can I determine which port was bound when setting port=0 for automatic port allocation?

**Comment by mhils** [comment, marked as answer]
You can a) parse mitmdump's stdout or b) write a custom addon that puts it wherever you want. :)

**Reply by willmckeandeb** [reply]
Hi @mhils, would you be able to show how to access the utilized port from inside an addon please? I tried reading `ctx.master.options.listen_port` from inside the `running` method, but it is always set to 0.

Many thanks

**Reply by mhils** [reply]
```
$ cat addr.py
from mitmproxy import ctx
def running():
    print(ctx.master.addons.get("proxyserver").listen_addrs())

$ mitmdump -s addr.py -p 0
[14:02:47.317] Loading script addr.py
[14:02:47.319] HTTP(S) proxy listening at *:56048.
[('::', 56048, 0, 0), ('0.0.0.0', 56048)]
```
